dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleAction.java

```diff
@@ -134,6 +137,18 @@ public class ProgramRuleAction extends BaseIdentifiableObject implements Metadat
    *   <li>sendmessage
    * </ul>
    */
+  @JsonIgnore private ProgramNotificationTemplate notificationTemplate;
+
+  /**
+   * The program notification template uid that will be triggered by the rule action. Used for:
+   *
+   * <ul>
+   *   <li>sendmessage
+   * </ul>
+   *
+   * @deprecated use {@link #notificationTemplate} instead
+   */
+  @Deprecated(since = "2.43")
```

Comment:
Why do we need to introduce `notificationTemplate` as an API field?
Would everything work fine now that we changed the DB to use the foreing key?
Can we have only `templateUid` as part of the API as before and make `notificationTemplate` @JsonIgnore?

Reply:
This is exactly what's happening

Reply:
> Why do we need to introduce `notificationTemplate` as an API field?
> 
> Would everything work fine now that we changed the DB to use the foreing key?
> 
> Can we have only `templateUid` as part of the API as before and make `notificationTemplate` @JsonIgnore?

Can we make it a foreign key without having it as field ?

Reply:
You are mixing 2 different approaches.
1. We can annotate `notificationTemplate` with @JsonIgnore, hence not using it in the API and use it in the DB and remove `templateuid` column. This approach do not involve any deprecation. The API remains the same and it will not be changed in the future.
2. For a transition period we accept either fields in the API `templateUid` and `notificationTemplate`, validating that if both are sent we are sending an error back, and specifing on `templateUid` when it is going to be removed with @Deprecated(forRemoval = true, since = "2.43"). The DB part is the same as the other, `templateuid` column is removed.

Reply:
We need to decide which one to use and follow just one approach